cmd/kubelet/app/server.go

```diff
@@ -595,6 +595,12 @@ func run(ctx context.Context, s *options.KubeletServer, kubeDeps *kubelet.Depend
 		// make a separate client for events
 		eventClientConfig := *clientConfig
 		eventClientConfig.QPS = float32(s.EventRecordQPS)
+
+		// If the value of EventRecordQPS is 0, there is no limit enforced
+		if int(eventClientConfig.QPS) == 0 {
```

Comment:
The field description say "If 0 will use DefaultQPS:"? Was defaulting happened somewhere above already?

Reply:
This comment means that if the QPS value is 0, `tokenBucketPassiveRateLimiter.qps` will be set to 5 when calling `NewForConfig`. 
Ref: https://github.com/kubernetes/kubernetes/blob/4f796c02f77fb95d42cd161ea663dd1bf05e372f/staging/src/k8s.io/client-go/rest/config.go#L355-L367

Reply:
so there is no way it will be `0`? Sorry, I didn't look too close in this logic, just asking for an explanation here

Reply:
Sorry, I misinterpreted what you said earlier.

Let me explain the changes made in this PR.

Because the field type of `eventRecordQPS` in `v1beta1.KubeletConfiguration` configuration is `*int32`.
https://github.com/kubernetes/kubernetes/blob/d8919343526597e0788a1efe133c70d9a0c07f69/staging/src/k8s.io/kubelet/config/v1beta1/types.go#L213-L217
https://github.com/kubernetes/kubernetes/blob/d8919343526597e0788a1efe133c70d9a0c07f69/pkg/kubelet/apis/config/v1beta1/defaults.go#L98-L100

If the user sets the value of `eventRecordQPS` to 0 in the kubelet configuration file, it will modify the `eventRecordQPS` in the internal `KubeletConfiguration` to 0 instead of using the default value. At this time, if we call `v1core.NewForConfig()`, the QPS will be set to 5.

https://github.com/kubernetes/kubernetes/blob/d8919343526597e0788a1efe133c70d9a0c07f69/cmd/kubelet/app/server.go#L697-L700
https://github.com/kubernetes/kubernetes/blob/4f796c02f77fb95d42cd161ea663dd1bf05e372f/staging/src/k8s.io/client-go/rest/config.go#L355-L367

This behavior is different from the comment we added for `eventRecordQPS` in the `v1beta1.KubeletConfiguration` configuration. We expect "**If 0, there is no limit enforced**". Therefore, this PR fixes this error.

The description of `--event-qps` and `--event-burst` modified in this PR seems are incorrect, it comes from: https://github.com/kubernetes/kubernetes/pull/117156. 
Because the args of kubelet will directly assign values to the internal `KubeletConfiguration`. So when `--event-qps` is 0, the `eventRecordQPS` in the internal `KubeletConfiguration` will also be set to 0. At this time, if we call `v1core.NewForConfig()`, the QPS will be set to 5, this does not conform to the description we modified.

Also incorrect is: https://github.com/kubernetes/kubernetes/blob/d8919343526597e0788a1efe133c70d9a0c07f69/cmd/kubelet/app/options/options.go#L483-L484

/cc @pacoxu 
Based on the above, do you think the description we modified for `event-qps` and `kube-api-qps` in the args an error? Or is it intentional? The modified args description is consistent with the description of the same field in the kubelet configuration file, but the behavior is inconsistent.

Reply:
Thank you for explanation. Is there any unit test can be added to test this?

Reply:
We may not be able to add unit tests or e2e tests for it, but I have already done manual testing locally. 

However, I am still curious: is the description of kubelet's args different from its behavior as we expected?

Reply:
> However, I am still curious: is the description of kubelet's args different from its behavior as we expected?

Who are you asking? I thought you tested in.

My motivation to change it is that I think other fields do this defaulting correctly. So this defaulting seems to be an inconsistency/bug. But if I am wrong - please let me know.

Reply:
> Who are you asking? I thought you tested in.

Sorry, my poor English made me express my meaning wrongly.

What I want to ask is: "Is this behavior where the description of kubelet's args is inconsistent with its comments what we expect?"

As I questioned in my previous reply:
>Based on the above, do you think the description we modified for event-qps and kube-api-qps in the args an error? Or is it intentional? The modified args description is consistent with the description of the same field in the kubelet configuration file, but the behavior is inconsistent.

I didn't wait for your reply or @pacoxu's reply in the comments. I thought this inconsistent behavior was what we expected, so I raised this question.

But then I had a conversation with @pacoxu and we unanimously felt that this behavior is an error.

There is also a similar error at: https://github.com/kubernetes/kubernetes/blob/d8919343526597e0788a1efe133c70d9a0c07f69/cmd/kubelet/app/options/options.go#L483-L484

I will revert the modification to the description of args and propose a new PR to fix and discuss it.